frontend/simple/js/state.js

```diff
@@ -5,22 +5,78 @@
 
 import Vue from 'vue'
 import Vuex from 'vuex'
-
-const S = require('string')
+import EventLog from './event-log'
 
 Vue.use(Vuex)
 
 // TODO: save only relevant state to localforage
 const state = {
+  logPosition: null,
   // TODO: this should be managed by Keychain, not here
-  loggedIn: false
+  loggedIn: false,
+  offset: []
 }
 
 // Mutations must be synchronous! Never call these directly!
 // http://vuex.vuejs.org/en/mutations.html
 const mutations = {
   LOGIN (state) { state.loggedIn = true },
-  LOGOUT (state) { state.loggedIn = false }
+  LOGOUT (state) { state.loggedIn = false },
+  UPDATELOG (state, hash) { state.logPosition = hash },
+  SETOFFSET (state, offset) {
+    if (!Array.isArray(offset)) {
+      throw new TypeError('Invalid Offset')
+    }
+    state.offset = offset
+  }
+}
+
+export const actions = {
+  apppendLog ({commit}, value) {
+    (async function () {
```

Comment:
It is very rare when async/await style is used like this (typically only as a top-level test).

This and the functions below should be rewritten as:

```js
async apppendLog ({commit}, value) {
  [... code ...]
}
```

Then there will also not be any reason to call `.bind(this)`